File: java/palindromo.java
import java.util.Scanner;

public class palindromo {
    public static void main(String[] args) {
         // Creamos un objeto Scanner para leer la entrada del usuario
        Scanner scanner = new Scanner(System.in);

        System.out.print("Ingresa un número: ");
        int numero = scanner.nextInt();
// Llamamos a la función esPalindromo y mostramos el resultado
        if (esPalindromo(numero)) {
            System.out.println(numero + " es un palíndromo.");
        } else {
            System.out.println(numero + " no es un palíndromo.");
        }
    }

    // Función que determina si un número es palíndromo
    public static boolean esPalindromo(int numero) {
        int original = numero;
        int reverso = 0;

        while (numero > 0) {
            int digito = numero % 10;
            reverso = reverso * 10 + digito;
            numero /= 10;
        }
        // Comparamos el número original con el reverso
        return original == reverso;
    }
}

File: cpp/palindromo.cpp
#include <iostream>
using namespace std;

// Función que verifica si un número es palíndromo
bool esPalindromo(int numero) {
    int original = numero;
    int invertido = 0;

    // Invertimos el número
    while (numero > 0) {
        int digito = numero % 10;
        invertido = invertido * 10 + digito;
        numero /= 10;
    }

    // Comparamos el número original con el invertido
    return original == invertido;
}

int main() {
    int numero;

    // Solicitar al usuario que ingrese un número
    cout << "Ingrese un numero: ";
    cin >> numero;

    // Verificar si es palíndromo y mostrar el resultado
    if (esPalindromo(numero)) {
        cout << "Es un numero palindromo." << endl;
    } else {
        cout << "No es un numero palindromo." << endl;
    }

    return 0; 
}
